HotSpot/View Controllers /CallViewController.m

```diff
@@ -0,0 +1,139 @@
+//
+//  CallViewController.m
+//  HotSpot
+//
+//  Created by drealin on 7/31/19.
+//  Copyright © 2019 aodemuyi. All rights reserved.
+//
+
+#import "CallViewController.h"
+
+#import "Parse/Parse.h"
+#import "CallManager.h"
+
+@interface CallViewController ()<UITextFieldDelegate, CallManagerDelegate>
+
+@property (nonatomic, weak) IBOutlet UIImageView *iconView;
+@property (nonatomic, assign, getter=isSpinning) BOOL spinning;
+
+@property (nonatomic, weak) IBOutlet UIButton *placeCallButton;
+@property (nonatomic, weak) IBOutlet UITextField *outgoingValue;
+@property (nonatomic, strong) UIAlertController* incomingAlertController;
+@property (weak, nonatomic) IBOutlet UIView *callControlView;
+@property (weak, nonatomic) IBOutlet UISwitch *muteSwitch;
+@property (weak, nonatomic) IBOutlet UISwitch *speakerSwitch;
+
+@property (strong, nonatomic) CallManager *callManager;
+
+@property BOOL visible;
+@end
+
+@implementation CallViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.callManager = [CallManager sharedCallManager];
+    self.callManager.delegate = self;
+    
+    [self toggleUIState:YES showCallControl:NO];
+    self.outgoingValue.delegate = self;
+    
+    if(self.listing) {
+        self.outgoingValue.text = self.listing.homeowner.objectId;
+    }
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+}
+
+
+# pragma mark - CallManagerDelegate
+
+- (void)toggleUIState:(BOOL)isEnabled showCallControl:(BOOL)showCallControl {
+    self.placeCallButton.enabled = isEnabled;
+    if (showCallControl) {
+        self.callControlView.hidden = NO;
+        self.muteSwitch.on = NO;
+        self.speakerSwitch.on = YES;
+    } else {
+        self.callControlView.hidden = YES;
+    }
+}
+
+- (void)setCallButtonTitle:(nullable NSString *)title {
+    [self.placeCallButton setTitle:title forState:UIControlStateNormal];
+}
+
+- (void)presentAlert:(UIAlertController *)alertController {
+    [self presentViewController:alertController animated:YES completion:nil];
+}
+
+- (void)startSpin {
+    if (!self.isSpinning) {
+        self.spinning = YES;
+        [self spinWithOptions:UIViewAnimationOptionCurveEaseIn];
+    }
+}
+
+- (void)stopSpin {
+    self.spinning = NO;
+}
+
+- (void)spinWithOptions:(UIViewAnimationOptions)options {
```

Comment:
we actually don't need that twilio icon right? we can customize our own UI? The way I envision it, I tap a call button on a profile page or something and the app makes the call. we probably don't even need a new view controller for this.